CONTRIBUTING.md

```diff
@@ -12,6 +11,57 @@ This monorepository contains the different subprojects for it:
                     i.e. as _lambda functions_
 
 
+## Terminology
+
+An Exocom application consists of a number of collaborating services.
+For example, the _todo-application_ consists of a _user_ service for storing user accounts,
+a _todo_ service for storing todo items,
+an _html_ service for exposing the application's functionality via an HTML UI,
+a _rest_ service for exposing it via a REST API,
+and a _chat_ service for exposing it via a chatbot.
+
+"User" and "todo" are the __role__ that the respective services play in the application.
+A role can be played by different __service types__.
+For example, you could use the _FacebookUserService_ service type
+to play the role of the _user_ service for your application in production,
+and you could use a _MockUserService_ to play the same role for local testing.
+You can also use the same service type for different roles,
+for example use a generic _MongoService_ as a quick-and-dirty prototype
+for both the _usersService_ and the _todoService_ role,
+since they both store data.
+
+To run the application, Exosphere creates an instance of the message bus,
+plus an instance of the respective service for each role.
+Or several instances for load balancing.
+These instances are called __clients__ (because they are clients of the message bus).
+Clients have unique names to identify them, for example "web #1", "web #2", etc.
+Clients communicate via the central __message bus__,
+played by the [Exocom-Server](exocom-server).
+
+The __application configuration__ file (`application.yml`) defines
+which roles the application has, and which _service type_ plays which role
```

Comment:
My understanding was that "service type" is defined in `service.yml`:
```
# users service service.yml

type: mongo # this is the service type
description: Manages user accounts in Exosphere applications
author: originate
```
and only the service role is defined in application.yml:
```
#application.yml

users: #service role
    location: ./exosphere-users-service
tweets: #service role
    location: ./exosphere-tweets-service
```